BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -108,6 +125,199 @@ public boolean onOptionsItemSelected(MenuItem item) {
+   *                     which is either PackageManager.PERMISSION_GRANTED
+   *                     or PackageManager.PERMISSION_DENIED.
+   */
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+                                         String[] permissions, int[] grantResults) {
+
+    switch (requestCode) {
+
+      case STORAGE_PERMISSION_CODE:
+        if (grantResults.length > 0
+            && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          createBibFile("Download", shelfName);
+          retrieveBibContent();
+          writeBibFile("Download", shelfName, bibExportContent);
+        } else {
+          Toast.makeText(getContext(), R.string.storage_permission_denied,
+              Toast.LENGTH_SHORT).show();
+        }
+        break;
+
+      default:
+    }
+  }
+
+  private void createBibFile(String folderName, String fileName) {
+    try {
+      String rootPath = Environment.getExternalStorageDirectory() + "/" + folderName + "/";
+      File root = new File(rootPath);
+
+      if (!root.exists()) {
+        root.mkdirs();
+      }
+
+      File file = new File(rootPath + fileName + ".bib");
+      if (file.exists()) {
+        file.delete();
+      }
+      file.createNewFile();
+
+      FileOutputStream out = new FileOutputStream(file);
+      out.flush();
+      out.close();
+
+      Toast.makeText(getContext(), getString(R.string.exported_file_stored_in) + '\n'
+          + "/" + folderName + "/" + fileName + ".bib", Toast.LENGTH_LONG).show();
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void retrieveBibContent() {
+
+    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
+    BookDao bd = new BookDao(dbHelper);
+    NoteDao nd = new NoteDao(dbHelper);
+
+    List<Long> bookIdsCurrShelf = bd.getAllBookIdsForShelf(shelfId);
+    List<Author> authorsCurrBook = new ArrayList<>();
+    List<Long> notesCurrBook = new ArrayList<>();
+    String allNotesCurrBook = "";
+    String authorNamesCurrBook = "";
+
+    //for each book in the current shelf
+    for (int i = 0; i < bookIdsCurrShelf.size(); i++) {
+      Book currBook;
+      Long currBookId = bookIdsCurrShelf.get(i);
+      authorsCurrBook = bd.getAllAuthorsForBook(currBookId);
+      notesCurrBook = nd.getAllNoteIdsForBook(currBookId);
+      currBook = bd.findById(currBookId);
+
+      /*
+      get the notes for the current book
+      and save the bib content in one string
+      */
+      for (int k = 0; k < notesCurrBook.size(); k++) {
+        String noteTextCurrBook = nd.findTextById(notesCurrBook.get(k));
+        allNotesCurrBook +=  "anote={" + noteTextCurrBook + "}," + '\n';
+      }
+
+      /*
+      get author's first and last name and include
+      the needed book data in a bib format
+      */
+      for (int u = 0; u < authorsCurrBook.size(); u++) {
+        authorNamesCurrBook = authorsCurrBook.get(u).getFirstName()
```

Comment:
Since the authorNamesCurrBook is initialized with an empty string, the first output would be:
"and Name1 FamilyName1 and Name2 FamilyName2" etc. I will check if there is more than one author.

Reply:
I suggest implementing exception for that? ;) We will definitely have examples with more than one author.

Reply:
I checked if there is more than one author. If yes then the names are concatenated.